easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -1858,83 +1860,72 @@ def det_installversion(version, toolchain_name, toolchain_version, prefix, suffi
-                if error_re.match(str(err)):
-                    if error_on_missing_easyblock:
-                        raise EasyBuildError("No software-specific easyblock '%s' found for %s", class_name, name)
-                elif error_on_failed_import:
-                    raise EasyBuildError("Failed to import %s easyblock: %s", class_name, err)
-                else:
-                    _log.debug("Failed to import easyblock for %s, but ignoring it: %s" % (class_name, err))
+            modulepath = get_module_path(easyblock)
+            _log.info(f"Derived full easyblock module path for {class_name}: {modulepath}")
+    else:
+        # if no easyblock specified, try to find if one exists
+        if name is None:
+            name = "UNKNOWN"
+        # The following is a generic way to calculate unique class names for any funny software title
+        class_name = encode_class_name(name)
+        # modulepath will be the namespace + encoded modulename (from the classname)
+        modulepath = get_module_path(class_name, generic=False)
 
-        if cls is not None:
-            _log.info("Successfully obtained class '%s' for easyblock '%s' (software name '%s')",
-                      cls.__name__, easyblock, name)
+        try:
+            __import__(modulepath, globals(), locals(), [''])
+        except ImportError as err:
+            _log.debug(f"Failed to import easyblock module '{modulepath}' (derived from easyconfig name): {err}")
+            # fallback to generic EasyBlock for easyconfigs
+            # without any easyblock specification or no specific easyblock
+            class_name = "EasyBlock"
+            modulepath = "easybuild.framework.easyblock"
+            _log.info(f"Using generic EasyBlock module for software '{name}'")
         else:
-            _log.debug("No class found for easyblock '%s' (software name '%s')", easyblock, name)
-
-        return cls
+            _log.debug(f"Module path '{modulepath}' found (derived from easyconfig name)")
 
+    try:
+        _log.debug(f"Importing easyblock class {class_name} from {modulepath}")
+        cls = get_class_for(modulepath, class_name)
+    except ImportError as err:
+        # when an ImportError occurs, make sure that it's caused by not finding the easyblock module,
+        # and not because of a broken import statement in the easyblock module
+        modname = modulepath.replace('easybuild.easyblocks.', '')
+        error_re = re.compile(rf"No module named '?.*/?{modname}'?")
+        _log.debug(f"Error regexp for ImportError on '{modname}' easyblock: {error_re.pattern}")
+        if error_re.match(str(err)):
+            if error_on_missing_easyblock:
+                raise EasyBuildError(f"Software-specific easyblock '{class_name}' not found for {name}")
+            else:
+                _log.debug(f"Easyblock for {class_name} not found, but ignoring it: {err}")
+        else:
```

Comment:
keep the `elif error_on_failed_import` here?

Reply:
Done in https://github.com/easybuilders/easybuild-framework/pull/4531/commits/eacd6e3b65f38c2b0505a2e5f873ad1eae72cc1a